bot/cogs/define_cog.py

```diff
@@ -0,0 +1,130 @@
+            for wordData in jsonData:
+                #Stems of the given word (Past Tense, Future Tense, Perfect Tense, etc.)
+                wordStems = wordData.get('meta',{}).get('stems',[])
+                
+                #Syllables of the given word
+                syllableData = wordData.get('hwi',{}).get('hw',"")
+                
+                #Pronunciation of the given word (With those weird letters)
+                pronunc = []
+                prsData = wordData.get('hwi',{}).get('prs',[])
+                
+                for soundData in prsData:
+                    pronunc.append(soundData.get('mw',""))
+                
+                #Type of the given word (Noun, Verb, Adjective, etc.)
+                wordType = wordData.get('fl',"")
+                
+                #Definitions of the given word
+                definitions = []
+                defData = wordData.get('shortdef',[])
+                
+                for defin in defData:
+                    definitions.append(defin)
+                
+                #Turn data into one long string (represents a page)
+                template = "Tenses: "
+                for s in range(len(wordStems)):
+                    template += wordStems[s]
+                    if (s != len(wordStems) - 1):
+                        template += ", "
+                template += "\n"
+
+                template += "Syllables: " + syllableData + "\n"
+
+                template += "Pronunciation: "
+                for s in range(len(pronunc)):
+                    template += pronunc[s]
+                    if (s != len(pronunc) - 1):
+                        template += ", "
+                template += "\n"
+
+                template += "Word Type: " + wordType + "\n"
+
+                template += "\n"
+                for s in range(len(definitions)):
+                    page = template + "Definition: " + definitions[s]
+                    page = page.replace("*"," | ")
+                    pages.append(page)
+
+        #If the word cannot be found, the JSON returns a list of other possible suggestions. 
+        elif (isinstance(jsonData[0], str)):
+            template = "Word not found, see also: "
+            for s in range(len(jsonData)):
+                template += jsonData[s]
+                if (s != len(jsonData) - 1):
+                    template += ", "
+    
+            pages = [template]
+
+        return pages
+    
+    @commands.command()
+    async def define(self, ctx, word):
+        """
+        Given a word, find its definition and any other relevant information
+        
+        USE: define <word>
+        EXAMPLE: define schadenfreude
+
+        For phrases, use underscores
+        EXAMPLE: define computer_science
+        """
+
+        actualWord = word.replace("_"," ")
+        word = word.replace("_","%20").lower()
+        url = f'https://www.dictionaryapi.com/api/v3/references/collegiate/json/{word}?key={self.api_key}'
```

Comment:
to match the rest of the bot lets make the api url a constant like API_URL at the top of the file, in module scope